source/adapters/cuda/command_buffer.cpp

```diff
@@ -1361,49 +1361,55 @@ UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferUpdateKernelLaunchExp(
     return UR_RESULT_ERROR_INVALID_VALUE;
   }
 
-  auto KernelCommandHandle = static_cast<kernel_command_handle *>(hCommand);
+  try {
+    auto KernelCommandHandle = static_cast<kernel_command_handle *>(hCommand);
 
-  UR_CHECK_ERROR(validateCommandDesc(KernelCommandHandle, pUpdateKernelLaunch));
-  UR_CHECK_ERROR(
-      updateKernelArguments(CommandBuffer->Device, pUpdateKernelLaunch));
-  UR_CHECK_ERROR(updateCommand(KernelCommandHandle, pUpdateKernelLaunch));
-
-  // If no work-size is provided make sure we pass nullptr to setKernelParams so
-  // it can guess the local work size.
-  const bool ProvidedLocalSize = !KernelCommandHandle->isNullLocalSize();
-  size_t *LocalWorkSize =
-      ProvidedLocalSize ? KernelCommandHandle->LocalWorkSize : nullptr;
-
-  // Set the number of threads per block to the number of threads per warp
-  // by default unless user has provided a better number.
-  size_t ThreadsPerBlock[3] = {32u, 1u, 1u};
-  size_t BlocksPerGrid[3] = {1u, 1u, 1u};
-  CUfunction CuFunc = KernelCommandHandle->Kernel->get();
-  auto Result = setKernelParams(
-      CommandBuffer->Context, CommandBuffer->Device,
-      KernelCommandHandle->WorkDim, KernelCommandHandle->GlobalWorkOffset,
-      KernelCommandHandle->GlobalWorkSize, LocalWorkSize,
-      KernelCommandHandle->Kernel, CuFunc, ThreadsPerBlock, BlocksPerGrid);
-  if (Result != UR_RESULT_SUCCESS) {
-    return Result;
-  }
+    UR_CHECK_ERROR(
```

Comment:
Why was this case in particular caught by the sanitizer, but the cases like [here](https://github.com/oneapi-src/unified-runtime/blob/ebc7760029503b367ae80653913544c75ee1cc8a/source/adapters/cuda/command_buffer.cpp#L819) where we have
`UR_CHECK_ERROR(getNodesFromSyncPoints(` outside of a try/catch aren't? Not that this change is wrong.

Reply:
This test was failing so the `throw` was caught. If a test passes and doesn't throw then this layer won't know that it *might* have thrown. I'll see if I can find any more cases where `try...catch` isn't being used but the most reliable way we can find such cases is if they fail.

Reply:
The Codeplay SYCL-Graph team have an internal ticket to go through the CUDA/HIP adapters and fix all these cases, so don't feel obliged to do here. Do you have a link/reference to which test "this test" is?

Reply:
I've caught some more `UR_CHECK_ERROR`s outside of `try catch` and fixed them in the last commit.

Reply:
> The Codeplay SYCL-Graph team have an internal ticket to go through the CUDA/HIP adapters and fix all these cases, so don't feel obliged to do here. Do you have a link/reference to which test "this test" is? Then can try write more ones like that when we make these changes.

There isn't actually a specific test set for this layer yet. The layer will just check if the adapter has thrown, and if so call `std::abort`. Potentially a test that would highlight all times `UR_CHECK_ERROR` is called outside of a try.. catch block is just to redefine the macro as

```c++
#define UR_CHECK_ERROR(Res) \
   throw;
```

Then this would be caught by the exception sanitiser layer and std::abort called with an intelligible error message.